### PR TITLE
pytrainer: 1.12.1 → 2.0.0

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
Finally GTK 3 support https://github.com/NixOS/nixpkgs/issues/18559

There are still some rough edges, which I reported upstream; also created a patch for modern webkitgtk support https://github.com/pytrainer/pytrainer/pull/174

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by rycee** [comment]
Ah, I was just working on upgrading to 2.0.0 yesterday. I'll have a look at this, I think your setup is nicer. Thanks!

File: pkgs/applications/misc/pytrainer/default.nix [hunk 1]
```diff
@@ -1,57 +1,88 @@
...
 
-  # This package contains no binaries to patch or strip.
-  dontPatchELF = true;
-  dontStrip = true;
+  LC_ALL = "en_US.UTF-8";
 
   checkPhase = ''
-    ${python.interpreter} -m unittest discover
+    env HOME=$TEMPDIR \
+      xvfb-run -s '-screen 0 800x600x24' \
```
Comment (rycee) [inline pkgs/applications/misc/pytrainer/default.nix]: Could have
```
TZ=Europe/Kaliningrad \
LC_ALL=en_US.UTF-8 \
```
here, then I believe you can skip the `LC_ALL` field above and perhaps also remove `fix-test-tz.patch`.

Comment (jtojnar) [reply pkgs/applications/misc/pytrainer/default.nix]: Thanks, this looks slightly cleaner.

**Comment by rycee** [comment]
Thanks for the update, I've tried it and everything seems to work great! Rebased to master in 99e72dc4133b600ba6a554cd2b0d890fa4f39bbf.

**Pull request closed without merging**